### PR TITLE
MGMT-3489 Update create ignition-configs in case HighAvailabilityMode is None

### PR DESCRIPTION
**Pull request by eranco74**

### CODE REVIEW

File: internal/ignition/ignition.go [hunk 1]
```diff
@@ -154,6 +154,10 @@ func (g *installerGenerator) Generate(ctx context.Context, installConfig []byte)
 		"OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE="+g.releaseImage,
 		"OPENSHIFT_INSTALL_INVOKER=assisted-installer",
 	)
+	// TODO: Might need to remove this once installer support bootstrap-in-place https://github.com/openshift/installer/pull/4482
+	if g.cluster.HighAvailabilityMode != nil && *g.cluster.HighAvailabilityMode == models.ClusterHighAvailabilityModeNone {
```
Comment (osherdp) [inline internal/ignition/ignition.go:158]: I think using ``swag.StringValue`` can make it prettier

File: internal/ignition/ignition.go [hunk 2]
```diff
@@ -234,7 +238,15 @@ func (g *installerGenerator) Generate(ctx context.Context, installConfig []byte)
 		g.log.Errorf("Failed to write file %s", installConfigPath)
 		return err
 	}
-
+	// TODO: remove this once installer support bootstrap-in-place https://github.com/openshift/installer/pull/4482
+	if g.cluster.HighAvailabilityMode != nil && *g.cluster.HighAvailabilityMode == models.ClusterHighAvailabilityModeNone {
+		// In case of single node rename bootstrap-in-place-for-live-iso.ign to bootstrap.ign
+		err = os.Rename(filepath.Join(g.workDir, "bootstrap-in-place-for-live-iso.ign"), filepath.Join(g.workDir, "bootstrap.ign"))
```
Comment (osherdp) [inline internal/ignition/ignition.go:244]: So why do we use other name in the beginning?
I think we should either assume installer to generate it with the new name and use as-is (so we have clarity of the scenario by looking at the file name)
Or, using ``bootstrap.ign`` everywhere

File: internal/ignition/ignition.go [hunk 3]
```diff
@@ -234,7 +238,15 @@ func (g *installerGenerator) Generate(ctx context.Context, installConfig []byte)
 		g.log.Errorf("Failed to write file %s", installConfigPath)
 		return err
 	}
-
+	// TODO: remove this once installer support bootstrap-in-place https://github.com/openshift/installer/pull/4482
+	if g.cluster.HighAvailabilityMode != nil && *g.cluster.HighAvailabilityMode == models.ClusterHighAvailabilityModeNone {
+		// In case of single node rename bootstrap-in-place-for-live-iso.ign to bootstrap.ign
```
Comment (osherdp) [inline internal/ignition/ignition.go:243]:
```suggestion
		// In case of single node - rename bootstrap ignition file
```

**Review by osherdp — CHANGES_REQUESTED** [review]

File: internal/ignition/ignition.go (hunk 2 above)
Comment (eranco74) [reply internal/ignition/ignition.go:244]: Currently, the installer requires the env and generates the ignition under a different name (bootstrap-in-place-for-live-iso.ign).
I hope later this week, the installer will converge towards (not requiring the env and generating the ignition with the same name) https://github.com/openshift/enhancements/pull/565#discussion_r545545030
Using it "as is" requires more changes (in assisted-installer and assisted-service getClusterFiles API).
So let's see what the openshift installer ends up duing and adjust later.

**Pull request merged**